src/AppBundle/Controller/BlogController.php

```diff
@@ -36,12 +36,18 @@ class BlogController extends Controller
     /**
      * @Route("/", name="blog_index")
      */
-    public function indexAction()
+    public function indexAction(Request $request)
     {
-        $em = $this->getDoctrine()->getManager();
-        $posts = $em->getRepository('AppBundle:Post')->findLatest();
```

Comment:
We should pass paginator and request objects into the repository to move out all this pagination logic from the controller. But I think better to keep it in the controller.